### PR TITLE
Warn the user that loading state files resets the current state

### PR DESCRIPTION
**Pull request by mathturtle**
I'm not sure if we want a 'Do not show this again' checkbox... if so this will need more work.

### CODE REVIEW

**Comment by cjh1** [comment]
@mathturtle This looks good, it does however warn even if the workspace is empty. We could just check with the ```ModuleManager``` and just issue the warning is any data sources are loaded?

**Comment by cryos** [comment]
This would be irritating for empty loads, it should minimally check for some loaded data sets. We may want to get a little cleverer in future, but that should be fine for now.

**Comment by mathturtle** [comment]
Updated to check if there are any data sources before showing the dialog.

File: tomviz/SaveLoadStateReaction.cxx [hunk 1]
```diff
@@ -79,6 +80,16 @@ bool SaveLoadStateReaction::loadState()
 
 bool SaveLoadStateReaction::loadState(const QString& filename)
 {
+  if (ModuleManager::instance().hasDataSources()) {
+    if (QMessageBox::No ==
+        QMessageBox::warning(
+          tomviz::mainWidget(), "Load State Warning",
+          "Current data and operators will be cleared when "
+          "loading a state file.  Proceed anyway?",
+          QMessageBox::Yes | QMessageBox::No, QMessageBox::No)) {
```
Comment (cjh1) [inline tomviz/SaveLoadStateReaction.cxx]: @alesgenova Pointed out to me earlier today that ```QMessageBox::warning(...)``` are marked as obsolete and should not be used in new code. We doesn't have to do it in this PR, but we should move away from using them.

Comment (cjh1) [reply tomviz/SaveLoadStateReaction.cxx]: Actually, it's only the signatures that take int's for the button types .... as you were.

**Comment by cjh1** [comment]
@mathturtle Just a few clang-format issues.

**Pull request merged**